core/miner/storage_fsm/impl/precommit_batcher_impl.hpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <atomic>
+#include <boost/multiprecision/cpp_int.hpp>
+#include <chrono>
+#include <libp2p/protocol/common/asio/asio_scheduler.hpp>
+#include <libp2p/protocol/common/scheduler.hpp>
+#include <map>
+#include <vm/actor/actor.hpp>
+#include "api/full_node/node_api.hpp"
+#include "const.hpp"
+#include "fsm/fsm.hpp"
+#include "miner/storage_fsm/precommit_batcher.hpp"
+#include "miner/storage_fsm/sealing_events.hpp"
+#include "miner/storage_fsm/types.hpp"
+#include "primitives/address/address.hpp"
+#include "primitives/types.hpp"
+#include "vm/actor/actor_method.hpp"
+#include "vm/actor/builtin/types/miner/sector_info.hpp"
+#include "vm/actor/builtin/v0/miner/miner_actor.hpp"
+
+namespace fc::mining {
+  using api::FullNodeApi;
+  using api::kPushNoSpec;
+  using api::SectorNumber;
+  using api::SectorPreCommitInfo;
+  using boost::multiprecision::cpp_int;
+  using fc::vm::actor::builtin::v0::miner::PreCommitBatch;
+  using libp2p::protocol::Scheduler;
+  using libp2p::protocol::scheduler::Handle;
+  using libp2p::protocol::scheduler::Ticks;
+  using libp2p::protocol::scheduler::toTicks;
+  using primitives::ChainEpoch;
+  using primitives::TokenAmount;
+  using primitives::address::Address;
+  using vm::actor::MethodParams;
+  using StorageFSM =
+      fsm::FSM<SealingEvent, SealingEventContext, SealingState, SectorInfo>;
+
+  class PreCommitBatcherImpl : public PreCommitBatcher {
+   public:
+    struct preCommitEntry {
+      preCommitEntry(primitives::TokenAmount number, SectorPreCommitInfo info);
+      preCommitEntry() = default;
+      primitives::TokenAmount deposit;
```

Comment:
You already use `using primitives::TokenAmount` above
```suggestion
     TokenAmount deposit;
```

Reply:
fixed